opensearch-operator/controllers/cluster_test.go

```diff
@@ -66,6 +67,14 @@ var _ = Describe("Cluster Reconciler", func() {
 				return true
 			}, timeout, interval).Should(BeTrue())
 		})
+
+		It("should apply the right cluster resources successfully", func() {
+			for _, nodePoolSpec := range OpensearchCluster.Spec.NodePools {
+				Expect(nodePoolSpec.Resources.Limits.Cpu()).Should(Equal("500m"))
```

Comment:
You'll need to do a conversion as you are comparing a Quantity and a string and go doesn't like that (same for the line below):
```suggestion
				Expect(nodePoolSpec.Resources.Limits.Cpu().String()).Should(Equal("500m"))
```